### DISCUSSION TITLE
How to force WinAPI version... WINVER or _WIN32_WINNT?

### DISCUSSION THREAD

**Discussion by jdx-john** [microsoft/vcpkg#20685] [General]
I wonder if some runtime errors we are having are down to mismatch in build between our code and vcpkg ports.

The docs (https://vcpkg.readthedocs.io/en/latest/users/triplets/) say we can use:

```
VCPKG_CXX_FLAGS
Sets additional compiler flags to be used when not using VCPKG_CHAINLOAD_TOOLCHAIN_FILE.

```

e.g.: `set(VCPKG_CXX_FLAGS -D_WIN32_WINNT=0x600)`

But a few quick questions:

1. Should I be using `WINVER` or `_WIN32_WINNT`? (I know this is a WinAPI question not vcpkg but I never knew the distinction)
2. How can I check my changes have 'stuck' / been applied in a build, to know my output has the specified changes (or not)?

**Comment by autoantwort** [comment]
1. According to [this](https://docs.microsoft.com/en-us/windows/win32/winprog/using-the-windows-headers?redirectedfrom=MSDN) you should set `NTDDI_VERSION` and `_WIN32_WINNT` 
2. When you run `vcpkg install --triplet=...` and it says everything is up to date, then it your changes should be respected. But Afaik _WIN32_WINNT does not change runtime behavior. It only enables more functions in the windows headers.

**Reply by jdx-john** [reply]
Unless there is a very recent change, if I modify my triplet file and re-run `install`, it simply tells me everything is already built. I've always found this a bit odd considering vcpkg uses quite strict hashing in other places. It would be nice it at least warned me. Maybe that would be a valid feature request?

On the issue I'm trying to understand I'm tearing my hair out so trying anything I can think of :)

**Reply by autoantwort** [reply]
In manifest mode that should work.

**Reply by jdx-john** [reply]
We cannot use MF unfortunately, or at least we haven't found a way to so far. Seems like that's all MS focus on now unfortunately.

**Reply by autoantwort** [reply]
What is your problem with MF? Multiple triplets at the same time are supported. I don't know a feature that is supported by classic mode but not by MF mode.

**Reply by jdx-john** [reply]
When I last looked multiple triplets were not supported... things change so fast! The specific case we have is a solution of ~100 projects using multiple build configurations (32/64 bit for example). When I looked into MF a few months back I just couldn't get that to work nicely.

**Reply by autoantwort** [reply]
Yeah we had the same problem at work. But #19767 changed where things get installed so you can now use MF mode with a solution and 32/64 projects.
